x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -188,7 +188,11 @@ public final class IndexPrivilege extends Privilege {
     public static final IndexPrivilege NONE = new IndexPrivilege("none", Automatons.EMPTY);
     public static final IndexPrivilege ALL = new IndexPrivilege("all", ALL_AUTOMATON, IndexComponentSelectorPredicate.ALL);
     public static final IndexPrivilege READ = new IndexPrivilege("read", READ_AUTOMATON);
-    public static final IndexPrivilege READ_CROSS_CLUSTER = new IndexPrivilege("read_cross_cluster", READ_CROSS_CLUSTER_AUTOMATON);
+    public static final IndexPrivilege READ_CROSS_CLUSTER = new IndexPrivilege(
+        "read_cross_cluster",
+        READ_CROSS_CLUSTER_AUTOMATON,
+        IndexComponentSelectorPredicate.ALL
```

Comment:
This is the key part: making the `read_cross_cluster` authorize both data and failure selectors for cross cluster search.